src/sidebar/store/create-store.js

```diff
@@ -26,28 +29,86 @@ const { createReducer, bindSelectors } = require('./util');
  * @param [any[]] middleware - List of additional Redux middlewares to use.
  */
 function createStore(modules, initArgs = [], middleware = []) {
-  // Create the initial state and state update function.
-  const initialState = Object.assign(
-    {},
-    ...modules.map(m => m.init(...initArgs))
-  );
-  const reducer = createReducer(...modules.map(m => m.update));
+  // Create the initial state and state update function. The "base"
+  // namespace is reserved for non-namespaced modules which will eventually
+  // be converted over.
+
+  // Namespaced objects for initial states.
+  const initialState = {
+    base: null,
+  };
+  // Namespaced reducers from each module.
+  const totalReducers = {
+    base: null,
+  };
+  // Namespaced selectors from each module.
+  const totalSelectors = {
+    base: {
+      selectors: {},
+      // Tells the bindSelector method to scope the store.
+      scopeSelector: true,
+    },
+  };
+
+  // Temporary list of non-namespaced modules used for createReducer.
+  const baseModules = [];
+
+  // Iterate over each module and prep each module's:
+  //    1. state
+  //    2. reducers
+  //    3. selectors
+  //
+  // Modules that have no namespace get dumped into the "base" namespace.
+  //
+  modules.forEach(module => {
+    if (module.namespace) {
+      initialState[module.namespace] = module.init(...initArgs);
+      totalReducers[module.namespace] = createReducer(module.update);
+      totalSelectors[module.namespace] = {
+        selectors: module.selectors,
+      };
+    } else {
+      // No namespace
+      totalSelectors.base.selectors = {
+        // Aggregate the selectors into a single "base" map
+        ...totalSelectors.base.selectors,
+        ...module.selectors,
+      };
+      initialState.base = {
+        ...initialState.base,
+        ...module.init(...initArgs),
+      };
+      baseModules.push(module);
+    }
+  });
+
+  // Create the base reducer for modules that are not opting in for namespacing
+  totalReducers.base = createReducer(...baseModules.map(m => m.update));
 
   // Create the store.
-  const defaultMiddleware = [
-    // The `thunk` middleware handles actions which are functions.
-    // This is used to implement actions which have side effects or are
-    // asynchronous (see https://github.com/gaearon/redux-thunk#motivation)
-    thunk,
-  ];
-  const enhancer = redux.applyMiddleware(...defaultMiddleware, ...middleware);
-  const store = redux.createStore(reducer, initialState, enhancer);
+  const store = configureStore({
+    reducer: totalReducers,
+    middleware: [thunk, ...middleware],
+    devTools: process.env.NODE_ENV !== 'production',
```

Comment:
I was about to write that there wasn't much point enabling the devtools as it didn't support applications that run in iframes. It looks like that [might have changed](https://github.com/zalmoxisus/redux-devtools-extension/issues/50) though. I'm happy to add this if it works and is useful, but see caveat about redux-starter-kit below.